src/mongo/connectToMongoClient.ts

```diff
@@ -15,6 +15,11 @@ export async function connectToMongoClient(connectionString: string, appName: st
         useNewUrlParser: true
     };
 
+    if (!!connectionString.match(/^mongodb:\/\/(localhost|127\.0\.0\.1)/)) {
```

Comment:
[this](https://github.com/microsoft/vscode-cosmosdb/blob/master/test/mongoConnectionStrings.test.ts#L50) seems to imply that "localhost" isn't always the first thing after "mongodb://"

Might be worth creating an `isLocalhost` function and adding unit tests against the same connection strings in those tests. Never hurts to be careful when dealing with security stuff